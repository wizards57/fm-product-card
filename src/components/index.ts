import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductCardHOCProps } from '../interfaces/interfaces';

import { ProductButtons } from './ProductButtons';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
// El orden las las importaciones y exportaciones si importa ssss
export { ProductButtons } from './ProductButtons';
export { ProductImage } from './ProductImage';
export { ProductTitle } from './ProductTitle';


export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons,

});

export default ProductCard;

