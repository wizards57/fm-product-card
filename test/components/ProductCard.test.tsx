import { ProductCard } from '../../src/components'
import renderer from 'react-test-renderer';
import React from "react";
import { product1 } from '../data/products';

const { act } = renderer;

describe('ProductCard', () => {

    test('edebe de mostrar el componente correctamente ', () => {
        
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    ()=>(
                        <h1>Prueba de renderizado</h1>
                    )
}
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
        
    }
    );

    test('should incrementar el contador', () => { 
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Prueba de renderizado</h1>
                            <span>{count}</span>
                            <button onClick={() => increaseBy(1)}></button>
                        </>
                    )
                }
            </ProductCard>
        );

        let tree = wrapper.toJSON();
        console.log(tree);
        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick();
            
        })

        tree = wrapper.toJSON();

        expect((tree as any).children[1].children[0]).toBe('1');

     })
    
    
})