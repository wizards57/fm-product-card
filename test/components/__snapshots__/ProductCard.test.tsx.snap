// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard edebe de mostrar el componente correctamente  1`] = `
<div
  className="productCard undefined "
>
  <h1>
    Prueba de renderizado
  </h1>
</div>
`;

exports[`ProductCard should incrementar el contador 1`] = `
<div
  className="productCard undefined "
>
  <h1>
    Prueba de renderizado
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
